test/client/actions/script.test.js

```diff
@@ -0,0 +1,132 @@
+
+const scriptQuestions = [
+  {
+    id: 4,
+    title: 'Age',
+    description: 'What is your age range?',
+    type: 'singleselect',
+    responses: '0-10,11-20,21-40,41-60,61+',
+    created_at: '2017-08-11T21:36:45.387Z',
+    updated_at: '2017-08-11T21:36:45.387Z',
+    script_id: 1,
+    question_id: 3,
+    sequence_number: 1
+  },
+  {
+    id: 5,
+    title: 'Hobbies',
+    description: 'What are your hobbies?',
+    type: 'multiselect',
+    responses: 'swimming,running,biking,sleeping,eating,weaving',
+    created_at: '2017-08-11T21:36:45.398Z',
+    updated_at: '2017-08-11T21:36:45.398Z',
+    script_id: 1,
+    question_id: 2,
+    sequence_number: 2
+  },
+  {
+    id: 6,
+    title: 'Ice Cream',
+    description: 'How do you feel about ice cream?',
+    type: 'paragraph',
+    responses: null,
+    created_at: '2017-08-11T21:36:45.514Z',
+    updated_at: '2017-08-11T21:36:45.514Z',
+    script_id: 1,
+    question_id: 1,
+    sequence_number: 3
+  }
+];
+let mock;
+let store;
+
+describe('script actions', () => {
+  describe('setScriptInfo', () => {
+    describe('type', () => {
+      it('should have the type "SET_SCRIPT_INFO"', () => {
+        expect(setScriptInfo().type).toEqual('SET_SCRIPT_INFO');
+      });
+    });
+    describe('payload', () => {
+      it('should pass on the payload we pass in', () => {
+        expect(setScriptInfo(newScript).payload.name).toEqual(newScript.name);
+        expect(setScriptInfo(newScript).payload.body).toEqual(newScript.body);
+        expect(setScriptInfo(newScript).payload.description).toEqual(newScript.description);
+      });
+    });
+  });
+  describe('setScriptQuestions', () => {
+    describe('type', () => {
+      it('should have the type "SET_SCRIPT_QUESTIONS"', () => {
+        expect(setScriptQuestions().type).toEqual('SET_SCRIPT_QUESTIONS');
+      });
+    });
+    describe('payload', () => {
+      it('should pass on the payload we pass in', () => {
+        expect(setScriptQuestions(scriptQuestions).payload).toEqual(scriptQuestions);
+      });
+    });
+  });
+  describe('fetchScript', () => {
+    beforeEach(() => {
+      mock = new MockAdapter(axios);
+      mock.onGet('/scripts/1').reply(200, newScript);
+      store = mockStore(initialState);
+    });
+    afterEach(() => {
+      mock.restore();
+    });
+    it('should add expected action to the store', () => {
+      const expectedAction = { type: SET_SCRIPT_INFO, payload: newScript };
+      return store.dispatch(fetchScript(1))
+        .then(() => {
+          const actions = store.getActions();
+          expect(actions[0]).toEqual(expectedAction);
+          expect(actions[0].payload).toEqual(newScript);
+        });
+    });
+  });
+  describe('fetchScriptQuestions', () => {
+    beforeEach(() => {
+      mock = new MockAdapter(axios);
+      mock.onGet('/scripts/1/scriptQuestions/').reply(200, scriptQuestions);
+      store = mockStore(initialState);
+    });
+    afterEach(() => {
+      mock.restore();
```

Comment:
I meant to ask you, but I ended up switching to .reset bc .restore was causing problems in subsequent tests.